src/nlr/Layer.cpp

```diff
@@ -823,6 +827,180 @@ void Layer::computeSymbolicBoundsForRelu()
     }
 }
 
+void Layer::computeSymbolicBoundsForSign()
+{
+    std::fill_n( _symbolicLb, _size * _inputLayerSize, 0 );
+    std::fill_n( _symbolicUb, _size * _inputLayerSize, 0 );
+
+    for ( unsigned i = 0; i < _size; ++i )
+    {
+        // Eliminate neurons are skipped
+        if ( _eliminatedNeurons.exists( i ) )
+        {
+            _symbolicLowerBias[i] = _eliminatedNeurons[i];
+            _symbolicUpperBias[i] = _eliminatedNeurons[i];
+
+            _symbolicLbOfLb[i] = _eliminatedNeurons[i];
+            _symbolicUbOfLb[i] = _eliminatedNeurons[i];
+            _symbolicLbOfUb[i] = _eliminatedNeurons[i];
+            _symbolicUbOfUb[i] = _eliminatedNeurons[i];
+
+            continue;
+        }
+
+        /*
+          There are two ways we can determine that a Sign has become fixed:
+
+          1. If the Sign's variable has been externally fixed
+          2. lbLb >= 0 (Positive) or ubUb <= 0 (Negative)
```

Comment:
@guykatzz if up<=0 then there is no fix because the function is non-continuous

Reply:
Not sure I understand. Fixed here means phase fixed - if ubOfUb is negative (should be <, not <=, in the comment), then the upper bound is negative, and the Sign is in the negative phase.